src/subcommands/verify_msrv.rs

```diff
@@ -29,15 +30,22 @@ pub fn run_verify_msrv_action<R: Output>(
 
     let version = manifest
         .minimum_rust_version()
-        .ok_or(CargoMSRVError::NoMSRVKeyInCargoToml(cargo_toml))?;
+        .ok_or(CargoMSRVError::NoMSRVKeyInCargoToml(cargo_toml.to_owned()))?;
     let version = version.try_to_semver(release_index.releases().iter().map(Release::version))?;
 
     let cmd = config.check_command_string();
     reporter.mode(ModeIntent::VerifyMSRV);
     let status = check_toolchain(version, config, reporter)?;
     report_verify_completion(reporter, &status, &cmd);
 
-    Ok(())
+    if status.is_success() {
+        Ok(())
+    } else {
+        Err(CargoMSRVError::SubCommandVerify(Error::VerifyFailed {
```

Comment:
By returning an Err, cargo-msrv will return with a non-zero exit code, as per https://github.com/foresterre/cargo-msrv/blob/a27a100a59ccf0eded1e1ff6eaddfd0f4a7c387a/src/bin/cargo-msrv.rs#L26-L34

The other changes are to support the two added test cases, which should ensure we'll keep returning a non-zero exit code on failure of verify (and a zero exit code for success) in the future.